From: Jinjie Ruan <ruanjinjie@huawei.com>
Subject: syscall.h: remove unused SYSCALL_MAX_ARGS
Date: Mon, 1 Dec 2025 20:06:32 +0800

Patch series "syscall: Cleanup and improve syscall_get_arguments()", v2.

Remove unused SYSCALL_MAX_ARGS and avoid memcpy() for
syscall_get_arguments() for arm64.


This patch (of 2):

SYSCALL_MAX_ARGS appears to have been unused since commit 32d92586629a
("syscalls: Remove start and number from syscall_set_arguments() args"),
so remove it.

Link: https://lkml.kernel.org/r/20251201120633.1193122-1-ruanjinjie@huawei.com
Link: https://lkml.kernel.org/r/20251201120633.1193122-2-ruanjinjie@huawei.com
Fixes: 32d92586629a ("syscalls: Remove start and number from syscall_set_arguments() args")
Signed-off-by: Jinjie Ruan <ruanjinjie@huawei.com>
Reviewed-by: Russell King (Oracle) <rmk+kernel@armlinux.org.uk>
Cc: Catalin Marinas <catalin.marinas@arm.com>
Cc: Charlie Jenkins <charlie@rivosinc.com>
Cc: Christian Zankel <chris@zankel.net>
Cc: "Dmitry V. Levin" <ldv@strace.io>
Cc: Helge Deller <deller@gmx.de>
Cc: Maciej W. Rozycki <macro@orcam.me.uk>
Cc: Marc Rutland <mark.rutland@arm.com>
Cc: Max Filippov <jcmvbkbc@gmail.com>
Cc: Steven Rostedt <rostedt@goodmis.org>
Cc: Thomas Gleinxer <tglx@linutronix.de>
Cc: Will Deacon <will@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/arm/include/asm/syscall.h    |    2 --
 arch/arm64/include/asm/syscall.h  |    2 --
 arch/xtensa/include/asm/syscall.h |    1 -
 3 files changed, 5 deletions(-)

--- a/arch/arm64/include/asm/syscall.h~syscallh-remove-unused-syscall_max_args
+++ a/arch/arm64/include/asm/syscall.h
@@ -77,8 +77,6 @@ static inline void syscall_set_nr(struct
 	}
 }
 
-#define SYSCALL_MAX_ARGS 6
-
 static inline void syscall_get_arguments(struct task_struct *task,
 					 struct pt_regs *regs,
 					 unsigned long *args)
--- a/arch/arm/include/asm/syscall.h~syscallh-remove-unused-syscall_max_args
+++ a/arch/arm/include/asm/syscall.h
@@ -92,8 +92,6 @@ static inline void syscall_set_nr(struct
 		(nr & __NR_SYSCALL_MASK);
 }
 
-#define SYSCALL_MAX_ARGS 7
-
 static inline void syscall_get_arguments(struct task_struct *task,
 					 struct pt_regs *regs,
 					 unsigned long *args)
--- a/arch/xtensa/include/asm/syscall.h~syscallh-remove-unused-syscall_max_args
+++ a/arch/xtensa/include/asm/syscall.h
@@ -61,7 +61,6 @@ static inline void syscall_set_return_va
 	regs->areg[2] = (long) error ? error : val;
 }
 
-#define SYSCALL_MAX_ARGS 6
 #define XTENSA_SYSCALL_ARGUMENT_REGS {6, 3, 4, 5, 8, 9}
 
 static inline void syscall_get_arguments(struct task_struct *task,
_
